src/OpenTelemetry.Instrumentation.Http/OpenTelemetry.Instrumentation.Http.csproj

```diff
@@ -28,4 +28,8 @@
     <PackageReference Include="OpenTelemetry" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Reference Include="System.Net.Http" Condition="'$(TargetFramework)' == '$(NetFrameworkMinimumSupportedVersion)'" />
+  </ItemGroup>
```

Comment:
Why this change?
Looks like this wasn't included before. 
https://www.nuget.org/packages/OpenTelemetry.Instrumentation.Http/1.6.0-beta.2#dependencies-body-tab

Reply:
Good eye there @TimothyMothra! Not totally sure. This baffled me and @utpilla a bit. We used to reference the 3.1 versions of these packages. Interestingly they have a `netstandard2.0` target but NOT a `net462` target. We are jumping to the .NET 8 RC versions which do have a `net462` target. For some reason, using that new dedicated `net462` target a bunch of these projects started complaining about missing `System.Net.Http`! My theory is that `HttpClient` is part of `netstandard2.0` but NOT `net462`. It seems like when the `netstandard2.0` bits were used the whole .NET Standard 2.0 API layer came with it including `HttpClient`. Best guess 😆 If you want to spend some time on it and try to figure it out I am very curious to know exactly what is going on but I decided not enough so to keep spending time on it.

Reply:
> It seems like when the netstandard2.0 bits were used the whole .NET Standard 2.0 API layer came with it including HttpClient.

Yes, that's indeed the case here. This screenshot is from dotPeek, which shows that the `netstandard2.0` bits for this package bring in `netstandard2.0` APIs:

![image](https://github.com/open-telemetry/opentelemetry-dotnet/assets/66651184/bd74913e-739e-4806-a2c2-d4e319f7ad6c)